Тимофеев Артем и Аршак Сагателян

spaghetti322 opened the pull request:
сразу для 3х мерного случая
P.s. анимация считается очень долго

spaghetti322 committed:
Add files via upload

spaghetti322 committed:
Update gravitation.py

vongostev commented:
Добрый день!
Доделайте комментарии.
Хранение координат неоптимально, создает лишний код.
Тесты не генерируют графики сразу, только по окончании тестов. Сообщения в тестах не очень информативны.
В качестве отчета лучше сделайте дополнительный ноутбук с отображением графиков.

vongostev commented:
Если очень долго считается, используйте параллелизацию и нумбу

spaghetti322 commented:
Пробовал ускорить, разницы почти нет. Анимация очень долго создается.

vongostev commented:
Не знаю, я просто воткнул `numba.njit` в 2 первые функции, получил ускорение в 25%. Если пройтись по циклам, можно еще раза в 2 ускорить. И, кстати, еще раз почитайте условие, у Варвары ошибки в полной энергии намного меньше.

vongostev commented:
> Пробовал ускорить, разницы почти нет. Анимация очень долго создается.

Про анимацию я сделал отдельный блокнот, смотрите в папке с заданием.

spaghetti322 committed:
Merge branch 'vongostev:main' into main

spaghetti322 committed:
Исправил многое

1. Энергия намного лучше сохраняется, если взять реальные значения (G=6.67e-11 и т.д.)
2. Добавил @numba.njit
3. Пофиксил скорость сохранения координат

spaghetti322 committed:
Add files via upload

vongostev commented:
Доброе утро!
В строке `self.coords = ([self.vec_P[0], self.vec_P[1], self.vec_P[2]])` что происходит? Зачем там и круглые, и квадратные скобки? Подумайте. В остальном неплохо.

spaghetti322 commented:
> Доброе утро! В строке `self.coords = ([self.vec_P[0], self.vec_P[1], self.vec_P[2]])` что происходит? Зачем там и круглые, и квадратные скобки? Подумайте. В остальном неплохо.

Там должно было стоять np.array, а вот куда оно пропало....

spaghetti322 committed:
Добавил np.array

vongostev commented:
Добрый вечер! Жду юнит-тестов и комментариев в коде (есть не все). Также попробуйте нумбу.

spaghetti322 committed:
Добавил свою часть unittest-ов. 

Тест на проверку орбиты делает Аршак, так что ждемс

Saga1308 committed:
Исправил детерминистический тест на тип траектории, закончил unittest
